gusto/physics.py

```diff
@@ -78,31 +79,26 @@ def __init__(self, state):
         w_sat = (w_sat1 /
                  (p * exp(w_sat2 * (T - T_0) / (T - w_sat3)) - w_sat4))
 
-        # correct w_sat to be positive
-        w_sat = interpolate(conditional(w_sat > 0, w_sat, 999.0), V)
-
         # make appropriate condensation rate
-        cond_rate = ((self.water_v - w_sat) /
-                     (dt * (1.0 + ((L_v ** 2.0 * w_sat) /
-                                   cp * R_v * T ** 2.0))))
+        dot_r_cond = ((self.water_v - w_sat) /
+                      (dt * (1.0 + ((L_v ** 2.0 * w_sat) /
+                                    (cp * R_v * T ** 2.0)))))
 
         # adjust cond rate so negative concentrations don't occur
-        cond_rate = interpolate(conditional(cond_rate < 0,
-                                conditional(-cond_rate > self.water_c / dt,
-                                            -self.water_c / dt,
-                                            cond_rate),
-                                conditional(cond_rate > self.water_v / dt,
-                                            self.water_v / dt,
-                                            cond_rate)), V)
-
-        self.water_v_new = Interpolator(self.water_v - dt * cond_rate, V)
-        self.water_c_new = Interpolator(self.water_c + dt * cond_rate, V)
+        self.lim_cond_rate = Interpolator(conditional(dot_r_cond < 0,
+                                                      max_value(dot_r_cond, - self.water_c / dt),
+                                                      min_value(dot_r_cond, self.water_v / dt)), Vt)
```

Comment:
So this says:
```python
if dot_r_cond < 0:
   dot_r_cond = max(dot_r_cond, -self.water_c/dt)
else:
   dot_r_cond = min(dot_r_cond, self.water_v/dt)
```

OK, and then the condensation levels are in water_v and water_c and they're updating like:

```
water_c = water_c + dt*cond_rate
water_v = water_v - dt*cond_rate
```

So here you're arranging that if cond_rate is too negative, then water_c goes to zero.  And equally, if cond_rate is too positive, that water_v goes to zero.

What I wonder is if you need all these intermediate fields?  Can you just not put the expressions in?  Or do you somehow need to arrange that the updates to water_c and water_v and theta are evaluated with the same value of cond_rate.